homeassistant/components/statistics/sensor.py

```diff
@@ -125,6 +125,7 @@ def __init__(
         self._entity_id = entity_id
         self.is_binary = self._entity_id.split(".")[0] == "binary_sensor"
         self._name = name
+        self._unique_id = f"{self._entity_id.replace('.', '_')}__statistics"
```

Comment:
Do we really want to use the source sensor's entity id as unique id, it means the unique_id is not stable if the source entity's entity_id is changed. There's nothing guaranteeing the source entity's ID is stable and not changed by the user.

Reply:
Kind of the point. In https://github.com/home-assistant/core/pull/49960 I would add the selected characteristic to the mix.

When the user changes the source entity (or the characteristic) the resulting statistics sensor tracks a new measure and should for example not continue with old measurements in the recorder. So far this made sense to me.

Given, the user could rename the entity_id of the source sensor. I would be good to use the unique_id of the source entity. Sound better?

Reply:
We mostly don't let the user specify a `unique_id` for integrations where users manually configure entities.
One exception is `automation`.

What problem do you really intend to solve by adding statistics sensors to the entity registry?

Reply:
Generally I thought it's desired to have a unique_ud for every entity. Secondly: 
![image](https://user-images.githubusercontent.com/2870104/140772015-905e01de-8580-4002-8633-2951fff5e2c1.png)
I want to be able to assign to sensor to a room etc.

Reply:
I'd suggest to mark this as draft and finish off #49960 first.